stream-chat-android-ui-components/src/main/res/values/styles.xml

```diff
@@ -478,6 +478,7 @@
         <item name="android:layout_height">@dimen/stream_ui_avatar_size_small</item>
         <item name="android:layout_marginTop">8dp</item>
         <item name="android:layout_marginEnd">8dp</item>
+        <item name="android:layout_marginStart">8dp</item>
```

Comment:
Not sure if this PR is ready to review (I was assuming so, as it's not a draft), but margins with the current approach are messed up in a lot of places.

![image](https://user-images.githubusercontent.com/12054216/129687458-7ac21398-f4b9-454f-b368-6326cab5580f.png)

I think the arbitrary predicate API in its current design is just not able to work correctly. 

The problem is that we have to know not only whether a message needs to show the avatar, but also whether it needs to show the margin to align with other messages in its message group that _do_ show an avatar.

Consider these two cases for messages that don't need to have avatars themselves:
- If the default predicate is used (avatar only for bottom position messages), all messages in non-bottom positions should still show the margin for the avatar.
- If we use a predicate that returns `false` in all cases, no messages should show the margin for the avatar, and they should just be lined up to the edge of the screen instead.

If the predicate can have arbitrary logic in it, you simply can't tell if a message needs the margin or not (consider `Random.nextBoolean()` for an extreme case).

We either have to also ask users if they want the margins to be present on the message (if they don't want the avatar), or have to restrict customizations options to hiding ours/theirs and showing for all / just bottom messages, or come up with some other approach that I didn't think of off the top of my head.

Reply:
You're right. I did my tests with true/false and forgot about that 😵‍💫. Let's make the margins customisable so the users can decide how to apply then.

Reply:
Now it is possible to configure the margin of the start and the end of `MessageListView`. The draw back is that the user need to understand that he/she needs to create space for the avatar, but the good side is that it is now fully customisable.

Reply:
I'll add the documentation for it tomorrow.